src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -205,16 +207,42 @@ class ADRController(private val encryption: Encryption,
                 listOf("name" to resourceFileName, "description" to artefact.second, "hash" to file.md5sum(),
                         "resource_type" to resourceType)
         val adr = adrClientBuilder.build()
+        val newDatasetHash = commonParameters[2].second
         return try
         {
             when (resourceId)
             {
                 null -> adr.postFile("resource_create", commonParameters + listOf("package_id" to id), filePart)
-                else -> adr.postFile("resource_patch", commonParameters + listOf("id" to resourceId), filePart)
+                else ->
+                {
+                    val fileHasNoChanges = uploadFileHasNoChanges(resourceId, newDatasetHash)
+                    if (fileHasNoChanges)
+                    {
+                        SuccessResponse(null).asResponseEntity()
```

Comment:
```suggestion
                        EmptySuccessResponse.asResponseEntity()
```
More generally I guess we need to be careful here: this response probably looks very different to that from `resource_patch`. But as long as the front-end isn't expecting access to resource `id` etc from response ([it doesn't currently](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/store/adr/actions.ts#L168)) then it's fine.